# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGL2ContextFramebuffers.cpp
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGL2ContextFramebuffers.cpp
--- a/dom/canvas/WebGL2ContextFramebuffers.cpp	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGL2ContextFramebuffers.cpp	Wed Jan 08 22:19:14 2020 +0000
@@ -66,13 +66,6 @@
                                     dstY0, dstX1, dstY1, mask, filter);
 }
 
-JS::Value WebGL2Context::GetFramebufferAttachmentParameter(
-    JSContext* cx, GLenum target, GLenum attachment, GLenum pname,
-    ErrorResult& out_error) {
-  return WebGLContext::GetFramebufferAttachmentParameter(cx, target, attachment,
-                                                         pname, out_error);
-}
-
 ////
 
 static bool ValidateBackbufferAttachmentEnum(WebGLContext* webgl,
@@ -113,8 +106,8 @@
 }
 
 bool WebGLContext::ValidateInvalidateFramebuffer(
-    GLenum target, const dom::Sequence<GLenum>& attachments,
-    ErrorResult* const out_rv, std::vector<GLenum>* const scopedVector,
+    GLenum target, const nsTArray<GLenum>& attachments,
+    std::vector<GLenum>* const scopedVector,
     GLsizei* const out_glNumAttachments,
     const GLenum** const out_glAttachments) {
   if (IsContextLost()) return false;
@@ -189,14 +182,14 @@
   return true;
 }
 
-void WebGL2Context::InvalidateFramebuffer(
-    GLenum target, const dom::Sequence<GLenum>& attachments, ErrorResult& rv) {
+void WebGL2Context::InvalidateFramebuffer(GLenum target,
+                                          const nsTArray<GLenum>& attachments) {
   const FuncScope funcScope(*this, "invalidateFramebuffer");
 
   std::vector<GLenum> scopedVector;
   GLsizei glNumAttachments;
   const GLenum* glAttachments;
-  if (!ValidateInvalidateFramebuffer(target, attachments, &rv, &scopedVector,
+  if (!ValidateInvalidateFramebuffer(target, attachments, &scopedVector,
                                      &glNumAttachments, &glAttachments)) {
     return;
   }
@@ -217,14 +210,14 @@
 }
 
 void WebGL2Context::InvalidateSubFramebuffer(
-    GLenum target, const dom::Sequence<GLenum>& attachments, GLint x, GLint y,
-    GLsizei width, GLsizei height, ErrorResult& rv) {
+    GLenum target, const nsTArray<GLenum>& attachments, GLint x, GLint y,
+    GLsizei width, GLsizei height) {
   const FuncScope funcScope(*this, "invalidateSubFramebuffer");
 
   std::vector<GLenum> scopedVector;
   GLsizei glNumAttachments;
   const GLenum* glAttachments;
-  if (!ValidateInvalidateFramebuffer(target, attachments, &rv, &scopedVector,
+  if (!ValidateInvalidateFramebuffer(target, attachments, &scopedVector,
                                      &glNumAttachments, &glAttachments)) {
     return;
   }